src/web-server/routes.js

```diff
@@ -1,6 +1,38 @@
-import { format404 } from './html-formatter';
+import { format404, format404AsJson } from './html-formatter';
+import fs from '../lib/fs';
 
 const wwwRegex = /\/www(\/.*)/;
+const wwwJsonRegex = /\/www\/json(\/.*)/;
```

Comment:
The route should be `/json/*` without any relation with `/www`

Reply:
Why, can you elaborate?

Reply:
IMO,  `/www/ ` is for browsing through our filesystem. Whereas, `/json/` is for retrieving various information to use throughout the whole app. So I believe there should be a clear distinction between the two.

Reply:
Isn't browsing is a matter of file retrieval and then content display with browser engine based on the content type? In case of `www/json` the file content is returned as a metadata which optionally can return file itself, too.

Reply:
I like the direction you took this, and want to go one step further.  It's pretty common these days for APIs to do the kind of thing you're doing here as a query string variable, consider:

`https://github.com/humphd/next/blob/master/Dockerfile`

vs

`https://github.com/humphd/next/blob/master/Dockerfile?raw=true`

My proposal is that we allow `?json=1` or `?json=true` or even just `?json` on the `/www/...` routes.  In other words:

* `/www/index.html` serves the contents of `/index.html`
* `/www/index.html?json=1` serves a JSON string with metadata about `/index.html` (size, last modified, etc)
* `/www/directory` serves an HTML dir listing of `/directory/*`
* `/www/directory?json=1` serves a JSON string, which contains an Array of metadata objects about each entry in the directory (i.e., what you get via `sh.ls()`)